src/pymatgen/core/periodic_table.py

```diff
@@ -523,7 +532,8 @@ def term_symbols(self) -> list[list[str]]:
         # Total ML = sum(ml1, ml2), Total MS = sum(ms1, ms2)
         TL = [sum(ml_ms[comb[e]][0] for e in range(v_e)) for comb in e_config_combs]
         TS = [sum(ml_ms[comb[e]][1] for e in range(v_e)) for comb in e_config_combs]
-        comb_counter = Counter(zip(TL, TS, strict=True))
+        # comb_counter: Counter = Counter(zip(TL, TS, strict=True))
```

Comment:
I'm not seeing any error here either: 

<img width="1226" alt="image" src="https://github.com/user-attachments/assets/43335be9-db0c-4aa7-bdc7-de54771c4fd4">

Reply:
Also fixed by running pre-commit on linux instead of mac, but for record-keeping the mypy errors on my mac for this line was `error: Unsupported left operand type for | ("type[dict[Any, Any]]")  [operator]` and `error: No overload variant of "zip" matches argument types "list[int]", "list[Union[float, Literal[0]]]", "bool"  [call-overload]`

Reply:
Yep that's again a python version issue, as `strict` is not available python <3.9, and the type overload doesn't exist.